spot-ml/ml_ops.py

```diff
@@ -0,0 +1,203 @@
+
+    YR=args.fdate[0:4]
+    MH=args.fdate[4:6]
+    DY=args.fdate[6:8]
+
+    #getting defaults for ConfigParser interpolation
+    DEFAULTS = vars(args)
+    DEFAULTS.update({'YR':YR,'MH':MH,'DY':DY})
+
+    ## prepare parameters pipeline stages
+    HPATH = conf.get('DEFAULT','HPATH',vars=DEFAULTS)
+    LPATH = conf.get('DEFAULT','LPATH',vars=DEFAULTS)
+    MAXRESULTS = conf.get('DEFAULT','MAXRESULTS')
+    DUPFACTOR = conf.get('DEFAULT','DUPFACTOR')
+    RAWDATA_PATH = conf.get(args.type,'{0}_PATH'.format(args.type.upper()), vars=DEFAULTS)
+    FEEDBACK_PATH = "{0}/{1}_scores.csv".format(conf.get('DEFAULT','LPATH',vars=DEFAULTS),args.type)
+    PREPROCESS_STEP = "{0}_pre_lda".format(args.type)
+    POSTPROCESS_STEP = "{0}_post_lda".format(args.type)
+    HDFS_WORDCOUNTS = "{0}/word_counts".format(HPATH)
+
+    ## paths for intermediate files
+    HDFS_DOCRESULTS = "{0}/doc_results.csv".format(HPATH)
+    LOCAL_DOCRESULTS = "{0}/doc_results.csv".format(LPATH)
+
+    HDFS_WORDRESULTS = "{0}/word_results.csv".format(HPATH)
+    LOCAL_WORDRESULTS = "{0}/word_results.csv".format(LPATH)
+
+    HDFS_SCORED_CONNECTS = "{0}/scores".format(HPATH)
+    HDFS_MODEL = "{0}/model".format(HPATH)
+
+    LDA_OUTPUT_DIR = "{1}/{1}".format(args.type,args.fdate)
+
+    #get nodes and create comma seperated list
+    NODES = conf.get('DEFAULT','NODES').split()
+    nodes_csl = ','.join(NODES)
+
+    cmd = "hdfs dfs -rm -R -f {0}".format(HDFS_WORDCOUNTS)
+    execute_cmd(cmd,logger)
+
+    cmd = "mkdir -p {0}".format(LPATH)
+    execute_cmd(cmd,logger)
+
+    # protect the flow_scores.csv file
+    cmd = "rm -f {0}/*.{dat,beta,gamma,other,pkl}".format(LPATH)
+    execute_cmd(cmd,logger)
+
+    cmd = "hdfs dfs -rm -R -f {0}".format(HDFS_SCORED_CONNECTS)
+    execute_cmd(cmd,logger)
+
+    # Add -p <command> to execute pre MPI command.
+    # Pre MPI command can be configured in /etc/spot.conf
+    # In this script, after the line after --mpicmd ${MPI_CMD} add:
+    # --mpiprep ${MPI_PREP_CMD}
+
+    if conf.get('mpi',"MPI_PREP_CMD"):
+        cmd = conf.get('mpi',"MPI_PREP_CMD")
+        execute_cmd(cmd,logger)
+
+    SPK_CONFIG = conf.get('spark','SPK_CONFIG')
+    SPK_DRIVER_MEM = conf.get('spark','SPK_DRIVER_MEM')
+    SPK_EXEC = conf.get('spark','SPK_EXEC')
+    SPK_EXEC_CORES = conf.get('spark','SPK_EXEC_CORES')
+    SPK_EXEC_MEM = conf.get('spark','SPK_EXEC_MEM')
+    SPK_EXEC_MEM_OVERHEAD = conf.get('spark', 'SPK_EXEC_MEM_OVERHEAD')
+    SPK_DRIVER_MAX_RESULTS = conf.get('spark', 'SPK_DRIVER_MAX_RESULTS')
+    SPK_DRIVER_MEM_OVERHEAD = conf.get('spark', 'SPK_DRIVER_MEM_OVERHEAD')
+    LDAPATH = conf.get('DEFAULT','LDAPATH')
+    LUSER = conf.get('DEFAULT','LUSER')
+    MPI_CMD = conf.get('mpi','MPI_CMD')
+    PROCESS_COUNT = conf.get('mpi','PROCESS_COUNT')
+    TOPIC_COUNT = conf.get('DEFAULT','TOPIC_COUNT')
+
+    if tol:
+        TOL = tol
+    else:
+        TOL = conf.get('DEFAULT','TOL')
+
+    #prepare options for spark-submit
+    spark_cmd = [
```

Comment:
Some of the values in spark_cmd and spark_extras are either modified or gone in spot branch.